### ISSUE TITLE
Series.astype(dict) with unnamed series raises KeyError

### ISSUE THREAD

**Issue by mvashishtha** [modin-project/modin#4317]
### System information
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**: macOS Monterey
- **Modin version** (`modin.__version__`): latest source
- **Python version**: 3.9.10

### Describe the problem

In pandas, you can pass to `astype` a dict with a single entry mapping from the series name to the new dtype. You can do the same in Modin, except that Modin always checks the internal column name `__reduced__` instead of the series name. So if we use `astype` on an unnamed series:

```python
import pandas
s = pandas.Series([1])
string_s = s.astype({s.name: str})
# shows <class 'str'>
print(type(string_s.iloc[0]))
```

pandas changes the type correctly, but if we do the same in Modin, we get `KeyError: 'Only a column name can be used for the key in a dtype mappings argument.'`. For the same reason, if we try to rename using the internal column name `__reduced__`:

```python
import modin.pandas as pandas
s = pandas.Series([1])
string_s = s.astype({"__reduced__": str})
# shows <class 'str'>
print(type(string_s.iloc[0]))
```

Modin works fine, but pandas raises `KeyError: 'Only the Series name can be used for the key in Series dtype mappings.'`

Modin needs to fix the check on the keys in the dict [here](https://github.com/modin-project/modin/blob/ee2440c53a1e3bd47736776e7c643f05c4a0db70/modin/pandas/base.py#L908) and make sure to map the `__reduced__` column in the query compiler.

**Comment by mvashishtha** [comment]
Note that once we've fixed #4314, we should also check that `s.astype(series)` with an unnamed Series `s` works.

**Comment by anmyachev** [comment]
The reproducer works on [master](https://github.com/modin-project/modin/commit/f32a9bda4be075e9c5c99252bfeeb23d5d135d4c).
